packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -621,28 +624,23 @@ export class EditorPage {
 	async unpublish(): Promise< void > {
 		await this.editorToolbarComponent.switchToDraft();
 
-		const frame = await this.getEditorHandle();
 		// @TODO: eventually refactor this out to a ConfirmationDialogComponent.
-		await frame.click( `div[role="dialog"] button:has-text("OK")` );
+		await this.editor.getByRole( 'button' ).getByText( 'OK' ).click();
 		// @TODO: eventually refactor this out to a EditorToastNotificationComponent.
-		await frame.waitForSelector(
-			'.components-editor-notices__snackbar :has-text("Post reverted to draft.")'
-		);
+		await this.editor.getByRole( 'button', { name: 'Dismiss this notice' } ).waitFor();
 	}
 
 	/**
-	 * Obtains the published article's URL from post-publish panels.
+	 * Obtains the published article's URL from the post-publish toast.
 	 *
-	 * This method is only able to obtain the published article's URL if immediately
-	 * preceded by the action of publishing the article *and* the post-publish panel
-	 * being visible.
+	 * This method is only able to obtain the published article's URL if the
+	 * post-publish toast is still visible on the page.
 	 *
+	 * @deprecated Please use the return value from `EditorPage.publish` where possible.
 	 * @returns {URL} Published article's URL.
 	 */
 	async getPublishedURLFromToast(): Promise< URL > {
-		const frame = await this.getEditorHandle();
-
-		const toastLocator = frame.locator( selectors.toastViewPostLink );
+		const toastLocator = this.editor.locator( selectors.toastViewPostLink );
```

Comment:
This method was originally removed, but I had to restore it to maintain support for the `privacy-testing` suite, in particular the `private page/post` spec.